services/repository/files/update.go

```diff
@@ -192,26 +189,92 @@ func CreateOrUpdateRepoFile(ctx context.Context, repo *repo_model.Repository, do
 		log.Error("%v", err)
 	}
 	defer t.Close()
+
 	hasOldBranch := true
-	if err := t.Clone(opts.OldBranch); err != nil {
+	err = t.Clone(opts.OldBranch)
+	if err != nil {
 		if !git.IsErrBranchNotExist(err) || !repo.IsEmpty {
 			return nil, err
 		}
 		if err := t.Init(); err != nil {
 			return nil, err
 		}
-		hasOldBranch = false
-		opts.LastCommitID = ""
 	}
+
+	filesOptions := map[string]*RepoFileOptions{}
+	for _, file := range opts.Files {
+		switch file.Operation {
+		case "create":
+			if err != nil {
+				hasOldBranch = false
+				opts.LastCommitID = ""
+			}
```

Comment:
I didn't get the point why this `hasOldBranch` logic was changed.

Before: hasOldBranch = false if `opts.OldBranch` doesn't exist.

Now: it's only set to false if there is a "create" operation.

And the new logic seems quite fragile, it uses the `err` variable from another code block. If some new code is inserted between these two code blocks (`err = someOtherOperation()`), the logic is totally broken, and it seems difficult to figure out.